src/components/Card/Card.tsx

```diff
@@ -1,69 +1,72 @@
 import { NavigationProp, useNavigation } from '@react-navigation/native';
-import { useState } from 'react';
+import { memo, useState } from 'react';
 import { View, TouchableOpacity, Image } from 'react-native';
+import { useSelector } from 'react-redux';
 import { RootStackParamList } from 'src/navigation';
+import { getIsLoggedIn } from 'src/store/selectors';
 import { TOMATO } from 'src/styles';
-import { AdressListingValues, IconName } from 'src/types';
+import { AccommodationListItem, IconName } from 'src/types';
 
 import { styles } from './Card.styles';
 import Icon from '../Icon/Icon';
 import Text from '../Text/Text';
+import showAlert from '../alert';
 
-interface CardProps {
-  id: string;
-  thumbnailUrl: string;
-  squareMeters: number;
-  numberOfRooms: number;
-  allowedNumberOfPeople: number;
-  price: number;
-  address: AdressListingValues;
-}
+type Props = {
+  item: AccommodationListItem;
+  onAddedToWishlist: (accommodationId: string) => void;
+  onRemoveFromWishlist: (accommodationId: string) => void;
+};
 
-const Card = ({
-  id,
-  thumbnailUrl,
-  squareMeters,
-  numberOfRooms,
-  allowedNumberOfPeople,
-  price,
-  address,
-}: CardProps) => {
+const Card = ({ item, onAddedToWishlist, onRemoveFromWishlist }: Props) => {
   const navigation = useNavigation<NavigationProp<RootStackParamList>>();
+  const isLoggedIn = useSelector(getIsLoggedIn);
   const [pressed, setPressed] = useState<boolean>(false);
 
-  const handlePressed = () => {
+  const handleToggleWishlist = () => {
+    if (!isLoggedIn) {
+      showAlert('warning', { message: 'You should login first!' });
+      return;
+    }
+
     setPressed(!pressed);
```

Comment:
setPressed(!pressed) should be invoked on function bottom, cuz 'set' may work asyncroniously